pyomo/contrib/pynumero/interfaces/nlp.py

```diff
@@ -19,1714 +21,500 @@
 where x \in R^{n_x} are the primal variables,
       x_L \in R^{n_x} are the lower bounds of the primal variables,
       x_U \in R^{n_x} are the uppper bounds of the primal variables,
-      g: R^{nx} \rightarrow R^{n_g} are general inequality constraints
-      bounded by g_L \in R^{n_g}, and g_U \in R^{n_g}
-      y: \in R^{n_g} are the dual variables corresponding to the
-      general inequality constraints
-
-The general inequalitites can be further query to obtain equality and
-strict inequality constraints. In such case the NLP problem considered
-has the following form
+      g: R^{n_x} \rightarrow R^{n_c} are constraints (combined 
+         equality and inequality)
+      
+The second interface (ExtendedNLP) extends the definition above and
+presents the NLP in the following form where the equality and
+inequality constraints are separated.
 
 minimize             f(x)
-subject to         c(x) = 0
-              d_L <= d(x) <= d_U
+subject to           h(x) = 0
+              q_L <= q(x) <= q_U
               x_L <=  x   <= x_U
 
-where c: R^{n_x} \rightarrow R^{n_c} are the equality constraints
-      d: R^{n_x} \rightarrow R^{n_d} are the inequality constraints
-      note that nc + nd = ng
-.. rubric:: Contents
+where x \in R^{n_x} are the primal variables,
+      x_L \in R^{n_x} are the lower bounds of the primal variables,
+      x_U \in R^{n_x} are the uppper bounds of the primal variables,
+      h: R^{n_x} \rightarrow R^{n_eq} are the equality constraints
+      q: R^{n_x} \rightarrow R^{n_ineq} are the inequality constraints
 
-"""
-import pyomo
-import pyomo.environ as aml
+Note: In the case of the ExtendedNLP, it is generally assumed that
+both the NLP and the ExtendedNLP interfaces are supported and
+consistent (ExtendedNLP inherits from NLP). For example, a user should
+be able to call set_duals or set_duals_eq and set_duals_ineq, or
+mix-and-match between evaluate_jacobian or evaluate_jacobian_eq and
+evaluate_jacobian_ineq.
 
-try:
-    import pyomo.contrib.pynumero.extensions.asl as _asl
-except ImportError as e:
-    print('{}'.format(e))
-    raise ImportError('Error importing asl while running nlp interface. '
-                      'Make sure libpynumero_ASL is installed and added to path.')
+.. rubric:: Contents
 
-from scipy.sparse import coo_matrix, csr_matrix
-import abc
-import numpy as np
-import tempfile
-import os
+"""
 import six
-import shutil
-
-__all__ = ['AmplNLP', 'PyomoNLP']
+import abc
 
+__all__ = ['NLP']
 
+#TODO: Decide if we need the condensing matrix stuff
```

Comment:
It looks like the condensing matrix stuff and mask generation was moved to utils so can this TODO comment be removed?